Tests/LibCore/TestLibCoreIODevice.cpp

```diff
@@ -76,3 +76,23 @@ TEST_CASE(file_get_read_position)
         EXPECT_EQ(offset, 0);
     }
 }
+
+TEST_CASE(file_lines_range)
+{
+    auto path = "long_lines.txt";
+    auto file_or_error = Core::File::open(path, Core::OpenMode::ReadOnly);
+    if (file_or_error.is_error()) {
+        warnln("Failed to open {}: {}", path, file_or_error.error());
+        VERIFY_NOT_REACHED();
+    }
+    auto file = file_or_error.release_value();
+    auto output_path = "/tmp/output.txt";
+    auto outfile_or_error = Core::File::open(output_path, Core::OpenMode::WriteOnly);
+    auto outputfile = outfile_or_error.release_value();
+    for (auto line : file->lines()) {
+        outputfile->write(line);
+        outputfile->write("\n");
+    }
+    file->close();
+    outputfile->close();
```

Comment:
Can we asset equality to validate the iteration actually worked as expected?

Reply:
Yeah, we probably should actually, would improve the original test as well.

Reply:
Are you planning on doing that with in this PR? Or later?

Reply:
Yeah, I'm planning to do it in this one, just had a bunch of other things to prioritize.

Reply:
Ok, This should be fixed in the newest version.